### PR TITLE
Adds AmCharts

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith Please Review
Signed-off-by: Mathew Zummo <mzummo@gmail.com>

### CODE REVIEW

**Comment by mlzummo** [comment]
NOT ready to merge

File: demos/config.js [hunk 1]
```diff
@@ -5,14 +5,65 @@ require.config({
...
+        
+        'amcharts': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/amcharts", "../bower_components/amcharts/dist/amcharts/amcharts"],
+        'amcharts/funnel': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/funnel", "../bower_components/amcharts/dist/amcharts/funnel"],
+        'amcharts/gauge': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/gauge", "../bower_components/amcharts/dist/amcharts/gauge"],
+        'amcharts/pie': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/pie", "../bower_components/amcharts/dist/amcharts/pie"],
+        'amcharts/radar': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/radar", "../bower_components/amcharts/dist/amcharts/radar"],
+        'amcharts/serial': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/serial", "../bower_components/amcharts/dist/amcharts/serial"],
+        'amcharts/xy': ["http://rawgit.com/arturgspb/bower-amcharts/v3.13.3/dist/amcharts/xy", "../bower_components/amcharts/dist/amcharts/xy"],
```
Comment (GordonSmith) [inline demos/config.js]: I think we should follow what amcharts does here:  http://www.amcharts.com/tutorials/amcharts-meets-requirejs/

Comment (mlzummo) [inline demos/config.js]: Error: Error: ENOENT, no such file or directory
wnloads\vis-2015-04-27\dist\amd\amcharts.serial.js'
In module tree:
    src/amcharts/Area
      src/amcharts/CommonSerial

Comment (mlzummo) [inline demos/config.js]: we could ... but we then need to modify gulp to say when we get to this section parse the "." as a directory

Comment (GordonSmith) [inline demos/config.js]: That can be fixed by adding the correct paths to the gulpfile...

I am working in this area anyway, so will accept this and then make the changes later.

File: gulpfile.js [hunk 2]
```diff
@@ -170,6 +172,7 @@ gulp.task("amd_bundles_src", function (done) {
             'colorbrewer': '../bower_components/colorbrewer/colorbrewer',
             'd3.layout.cloud': '../bower_components/d3-cloud/d3.layout.cloud',
             'font-awesome': '../bower_components/font-awesome/css/font-awesome',
+            'amcharts': '../bower_components/amcharts/dist/amcharts',
```
Comment (GordonSmith) [inline gulpfile.js]: FYI - The paths here should be the same as the config.js (but don't change it now, as I am removing this list from the gulp file.

**Comment by mlzummo** [comment]
ok, I'm not sure where i needed to modify the gulp file, i messed with a few configurations without any success so ill take a look after you change the gulp file.

File: src/amcharts/Area.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    Area.prototype.update = function(domNode, element) {
+        CommonSerial.prototype.update.apply(this, arguments);
+        
+        this.updateChartOptions();
+        
+        this._chart.validateNow();
+        this._chart.validateData();
+    };
+
+    Area.prototype.testData = function() {
```
Comment (GordonSmith) [inline src/amcharts/Area.js]: This is not needed as its the same as INDChart

Comment (mlzummo) [inline src/amcharts/Area.js]: i see a few other areas that need the same fix, will fix too ..

**Comment by mlzummo** [comment]
updated

File: src/amcharts/Bar.css [hunk 4]
```diff
@@ -0,0 +1,5 @@
+div.amcharts_charts_Bar {
+	width		: 100%;
+	height		: 500px;
+	font-size	: 11px;
+}
```
Comment (GordonSmith) [inline src/amcharts/Bar.css]: This looks wrong?

File: src/amcharts/Bar.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+"use strict";
+(function(root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "./CommonSerial", "amcharts/serial", "../chart/INDChart", "css!./Bar"], factory);
+    } else {
+        root.amcharts_Bar = factory(root.d3, root.amcharts_CommonSerial, root.amcharts, root.chart_INDChart);
```
Comment (GordonSmith) [inline src/amcharts/Bar.js]: "amcharts_CommonSerial" looks wrong

File: src/amcharts/Bubble.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        this._tag = "div";
+        
+        this._type = "Bubble";
+        this._gType = "column";
+    };
+    Bubble.prototype = Object.create(CommonXY.prototype);
+
+    Bubble.prototype.implements(INDChart.prototype);
+
+    Bubble.prototype.testData = function() {
```
Comment (GordonSmith) [inline src/amcharts/Bubble.js]: Its better to default to the INDChart test data (for when it changes in the future, to add more tests for things like annotations).

File: src/amcharts/CommonFunnel.js [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    
+    CommonFunnel.prototype.publish("dataDateFormat", null, "string", "");
+    
+    CommonFunnel.prototype.publish("3dDepth", 0, "number", "3D Depth (px)");
+    CommonFunnel.prototype.publish("3dAngle", 0, "number", "3D Angle (Deg)");
+    
+    CommonFunnel.prototype.updateChartOptions = function() {
+        var context = this;
+
+        this._chart.pathToImages = "//cdn.amcharts.com/lib/3/images/";
```
Comment (GordonSmith) [inline src/amcharts/CommonFunnel.js]: Is this needed?

**Comment by GordonSmith** [comment]
Two comments:
1.  Is the reference to the CDN server for images needed?  
2.  Can you remove the testData functions that are similar to the one in INDChart and for the one that are not similar can you remove the reference to INDChart in the define and in the widget (as they are a different data "shape").

**Comment by mlzummo** [comment]
closed, pulled via new branch: https://github.com/hpcc-systems/Visualization/pull/335

**Pull request closed without merging**